### PR TITLE
feat(rome_js_analyze): noGlobalObjectCalls, no-obj-calls #3977

### PR DESCRIPTION
**Pull request by denbezrukov**
## Summary

Fix https://github.com/rome/tools/issues/3977

## Test Plan

`cargo test`

## Documentation

- [ ] The PR requires documentation
- [ ] I will create a new PR to update the documentation

### CODE REVIEW

File: crates/rome_js_analyze/src/analyzers/nursery/no_global_object_calls.rs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            QueryNode::JsNewExpression(expression) => expression.callee(),
+            QueryNode::JsCallExpression(expression) => expression.callee(),
+        }
+    }
+}
+
+const NON_CALLABLE_GLOBALS: [&str; 5] = ["Atomics", "JSON", "Math", "Reflect", "Intl"];
+
+fn is_non_callable_globals(text: &str) -> bool {
+    NON_CALLABLE_GLOBALS.contains(&text)
```
Comment (ematipico) [inline crates/rome_js_analyze/src/analyzers/nursery/no_global_object_calls.rs:180]: nit: there have been some interesting discoveries, and we found out that [using a `match` yields a better binary search](https://github.com/rome/tools/pull/3832#discussion_r1030499121) code, thanks to LLVM. We might want try that approach

File: crates/rome_js_analyze/src/analyzers/nursery/no_global_object_calls.rs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+    fn diagnostic(ctx: &RuleContext<Self>, token: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        let name = token.text_trimmed();
+
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.syntax().text_trimmed_range(),
+            markup! {
+                ""<Emphasis>{name}</Emphasis>" is not a function."
```
Comment (ematipico) [inline crates/rome_js_analyze/src/analyzers/nursery/no_global_object_calls.rs:157]: I think the first `""` is not needed

**Review by ematipico — APPROVED** [review]

File: crates/rome_js_analyze/src/analyzers/nursery/no_global_object_calls.rs (hunk 1 above)
Comment (denbezrukov) [reply crates/rome_js_analyze/src/analyzers/nursery/no_global_object_calls.rs]: Fixed

File: crates/rome_js_analyze/src/analyzers/nursery/no_global_object_calls.rs (hunk 2 above)
Comment (denbezrukov) [reply crates/rome_js_analyze/src/analyzers/nursery/no_global_object_calls.rs]: Indeed, 
Removed

**Pull request merged**